### PR TITLE
[WAIT] PHPC-1274: Reset libmongoc client after forking

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-1274

This is _not_ suitable for merging. Just opening so we have a POC to look at.

### CODE REVIEW

**Comment by derickr** [comment]
> This is suitable for merging.

``s/suitable/not suitable``, I suppose?

**Comment by jmikola** [comment]
> `s/suitable/not suitable`, I suppose?

Good catch :)

**Comment by jmikola** [comment]
Note: the current version of this PR scatters pid checks throughout the driver. Although calls to `getpid()` [may be cached](https://stackoverflow.com/a/41783404/162228) for some versions of glibc, I think the overhead might still be a concern.

An alternative approach could entail exposing the API for [`mongoc_client_reset()`](http://mongoc.org/libmongoc/current/mongoc_client_reset.html) behind our own method on the Manager class, which does its own pid checking, and instruct users to manually call that after forking.

File: php_phongo.h [hunk 1]
```diff
@@ -212,6 +212,13 @@ zend_bool phongo_writeconcernerror_init(zval* return_value, bson_t* bson TSRMLS_
 #define PHONGO_ZVAL_CLASS_OR_TYPE_NAME(zv) (Z_TYPE(zv) == IS_OBJECT ? ZSTR_VAL(Z_OBJCE(zv)->name) : zend_get_type_by_const(Z_TYPE(zv)))
 #define PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(zvp) PHONGO_ZVAL_CLASS_OR_TYPE_NAME(*(zvp))
 
+#define PHONGO_CHECK_CLIENT_PID(client, created_by_pid) \
```
Comment (sgolemon) [inline php_phongo.h]: This name could be more descriptive of what it does.
Perhaps: PHONGO_CLIENT_RESET_IF_CHIlD_PID(...)

**Comment by jmikola** [comment]
Standalone SSL failure is pending [CDRIVER-3116](https://jira.mongodb.org/browse/CDRIVER-3116).

**Comment by alcaeus** [comment]
Closing after discussion summarised in [this comment](https://jira.mongodb.org/browse/PHPC-1274?focusedCommentId=2442888&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2442888).

**Pull request closed without merging**